ACPI: PCI: introduce sleep to avoid occasional early kernel hang

Upgrading from v4.19 to v5.4 has resulted in an occasional early
kernel hang on a specific HW platform when no logs go to console,
which is the case with the quiet setting in /proc/cmdline. While no
fix to this has been identified, a workaround is to introduce a delay
to allow the hardware to be ready before scanning the root bridge.

Signed-off-by: Mike Manning <mmanning@vyatta.att-mail.com>
---
 drivers/acpi/pci_root.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/acpi/pci_root.c
+++ b/drivers/acpi/pci_root.c
@@ -20,6 +20,7 @@
 #include <linux/slab.h>
 #include <linux/dmi.h>
 #include <linux/platform_data/x86/apple.h>
+#include <linux/delay.h>
 #include <acpi/apei.h>	/* for acpi_hest_init() */
 
 #include "internal.h"
@@ -590,6 +591,12 @@
 	negotiate_os_control(root, &no_aspm, is_pcie);
 
 	/*
+	 * TBD: Fix needed so as to remove this workaround:
+	 * Avoid a race condition by delaying scan until all hardware is ready.
+	 */
+	msleep_interruptible(100);
+
+	/*
 	 * TBD: Need PCI interface for enumeration/configuration of roots.
 	 */
 
